privatelink-with-network-firewall.yaml

```
@@ -0,0 +1,818 @@
+    Properties:
+      Tags:
+        - Key: Name
+          Value: !Join ["", [!Ref pRosaClusterName, "-egress-igw"]]
+  rInternetGatewayEgressAttachment:
+    Type: AWS::EC2::VPCGatewayAttachment
+    Properties:
+      VpcId:
+        Ref: rEgressVpc
+      InternetGatewayId:
+        Ref: rInternetGateway
+  # NAT gateway
+  rElasticIP:
+    Type: AWS::EC2::EIP
+    Properties:
+      Domain: vpc
+      Tags:
+        - Key: Name
+          Value: !Join ["", [!Ref pRosaClusterName, "-egress-public-nat-eip"]]
+  rNATGateway:
+    Type: AWS::EC2::NatGateway
+    Properties:
+      AllocationId:
+        Fn::GetAtt:
+          - rElasticIP
+          - AllocationId
+      SubnetId:
+        Ref: rEgressVpcPublicSubnet
+      Tags:
+        - Key: Name
+          Value: !Join ["", [!Ref pRosaClusterName, "-egress-public-nat"]]
+
+  # Network Firewall
+  rEgressFirewall:
+    Type: AWS::NetworkFirewall::Firewall
+    Properties:
+      FirewallName: !Join ["", [!Ref pRosaClusterName, "-network-firewall"]]
+      VpcId: !Ref rEgressVpc
+      SubnetMappings:
+        - SubnetId: !Ref rEgressVpcPrivateSubnet
+      DeleteProtection: false
+      FirewallPolicyChangeProtection: false
+      SubnetChangeProtection: false
+      Tags:
+        - Key: "Name"
+          Value: !Join ["", [!Ref pRosaClusterName, "-network-firewall"]]
+      Description: Network Firewall for Egress VPC
+      FirewallPolicyArn: !Ref rEgressFirewallPolicy
+
+  rEgressFirewallPolicy:
+    Type: AWS::NetworkFirewall::FirewallPolicy
+    Properties:
+      FirewallPolicyName: !Join ["", [!Ref pRosaClusterName, "-firewall-policy"]]
+      FirewallPolicy:
+        StatelessDefaultActions:
+          - "aws:forward_to_sfe"
+        StatelessFragmentDefaultActions:
+          - "aws:forward_to_sfe"
+        StatefulRuleGroupReferences:
+          - ResourceArn: 
+              Ref: "rEgressFirewallRuleGroup1"
+          - ResourceArn: 
+              Ref: "rEgressFirewallRuleGroup2"
+          - ResourceArn: 
+              Ref: "rEgressFirewallRuleGroup3"
+      Description: Stateful rule group for Network Firewall
+      Tags:
+        - Key: "Name"
+          Value: !Join ["", [!Ref pRosaClusterName, "-network-firewall-policy"]]
+
+  rEgressFirewallRuleGroup1:
+    Type: "AWS::NetworkFirewall::RuleGroup"
+    Properties:
+      Capacity: "100"
+      RuleGroupName: !Join ["", [!Ref pRosaClusterName, "-firewall-rule-group1"]]
+      Type: "STATEFUL"
+      Tags:
+        - Key: "Name"
+          Value: !Join ["", [!Ref pRosaClusterName, "-firewall-rule-group1"]]
+      RuleGroup:
+        RuleVariables:
+          IPSets:
+            HOME_NET:
+              Definition:
+                - "10.0.0.0/16"
+                - "10.1.0.0/16"
+        RulesSource:
+          RulesSourceList:
+            TargetTypes:
+              - "TLS_SNI"
+              - "HTTP_HOST"
+            Targets:
+              - "registry.redhat.io"
+              - ".quay.io"
```

Missing `quay.io` (in addition to `.quay.io`):
```suggestion
              - ".quay.io"
              - "quay.io"
```

---

This gives error if I include quay.io 
`Resource handler returned message: "Targets has duplicate values, parameter: [quay.io], context: RulesSource.RulesSourceList.Targets[2] (Service: NetworkFirewall, Status Code: 400, Request ID: 8dabff27-e8e4-40fa-aba4-e5928bddc0d0)" (RequestToken: b189b206-f146-e004-56a9-1a9d75ac1dc9, HandlerErrorCode: InvalidRequest)`

---

Fixed